src/validation.cpp

```diff
@@ -1731,19 +1731,23 @@ DisconnectResult CChainState::DisconnectBlock(const CBlock& block, const CBlockI
     return fClean ? DISCONNECT_OK : DISCONNECT_UNCLEAN;
 }
 
-void static FlushBlockFile(bool fFinalize = false)
+static void FlushUndoFile(int block_file, bool finalize = false)
 {
-    LOCK(cs_LastBlockFile);
+    FlatFilePos undo_pos_old(block_file, vinfoBlockFile[block_file].nUndoSize);
+    if (!UndoFileSeq().Flush(undo_pos_old, finalize)) {
+        AbortNode("Flushing undo file to disk failed. This is likely the result of an I/O error.");
+    }
+}
 
+static void FlushBlockFile(bool fFinalize = false)
+{
+    LOCK(cs_LastBlockFile);
     FlatFilePos block_pos_old(nLastBlockFile, vinfoBlockFile[nLastBlockFile].nSize);
-    FlatFilePos undo_pos_old(nLastBlockFile, vinfoBlockFile[nLastBlockFile].nUndoSize);
-
-    bool status = true;
-    status &= BlockFileSeq().Flush(block_pos_old, fFinalize);
-    status &= UndoFileSeq().Flush(undo_pos_old, fFinalize);
-    if (!status) {
+    if (!BlockFileSeq().Flush(block_pos_old, fFinalize)) {
         AbortNode("Flushing block file to disk failed. This is likely the result of an I/O error.");
     }
+    // the undo file is actually finalized at a later stage, but we do want to flush it along with the block file
+    if (!fFinalize) FlushUndoFile(nLastBlockFile);
```

Comment:
We want to flush the rev file along with the block file, but we are not going to do that if `fFinalize` is `true`. What about calling `FlushUndoFile(nLastBlockFile)` unconditionally here?

Reply:
There's basically no reason to do a non-final flush of the undo file, except on shutdown (which happens now). This is the same for the block file (since the case you are arguing is for a finalized flush).

Reply:
Ok, then this turns into a nit, feel free to ignore: maybe update the comment to be in line with the code: "we do want to flush it along with the block file _in non-finalizing flushes_".

Reply:
I didn't see this comment until now. I'll update the code in the next push.